src/Infrastructure/Config/src/ESMF_Config.F90

```diff
@@ -3283,15 +3367,20 @@ subroutine ESMF_ConfigParseAttributes( config, unique, rc )
       do while ( i .lt. config%cptr%nbuf )
 
         ! get next line from buffer
-        j = i + index_(config%cptr%buffer(i:config%cptr%nbuf), EOL) - 1
-        this_line = config%cptr%buffer(i:j)
+        j = index_(config%cptr%buffer(i:config%cptr%nbuf), EOL)
```

Comment:
@oehmke 
`i` is the beginning index of a character string
`j` is the ending index of a character string.
`index_` searches for the next EOL (end of line) character.
`%nbuf` is the used buffer space, the last character in used buffer space is supposed to be a EOB (end of buffer) character

if EOL is found then this_label is everything from `i` to `EOL`, inclusive. This matches the old code but the old code assumed that you would always find EOL, which in theory is true.
if a EOL isn't found then this_label is everything from `i` to `EOB - 1` (so excluding the EOB character) and previously it would actually be `i` to `i - 1`, which would be an undefined string.

Reply:
Ah, so the -1 is to avoid the EOB. Looking at the case where it finds the EOL, then the max of that would be %nbuf-1 too, because the last character is EOB not EOL. Ok, that all makes sense to me now. Nice work. I'm fine with you merging this now.